src/com/googlecode/jmxtrans/model/output/GraphiteWriter.java

```diff
@@ -101,6 +101,11 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 							log.debug("Graphite Message: {}", line);
 							writer.write(line);
 							writer.flush();
```

Comment:
Nope, the `checkError()` will call `flush()` so you are good with just calling `checkError()`.